source/extensions/common/utility.h

```diff
@@ -43,9 +43,37 @@ class ExtensionNameUtil {
   }
 
   /**
-   * Checks the status of deprecated extension and either generates a log message or throws.
+   * Checks the status of deprecated extension names and generates a log message.
+   *
+   * @param extension_type absl::string_view that contains the extension type, for logging
+   * @param deprecated_name absl::string_view that contains the deprecated name, for logging
+   * @param canonical_name absl::string_view that contains the canonical name, for logging
+   * @param runtime Runtime::Loader used to determine if deprecated extension names are allowed.
+   * @return true if deprecated extensions are allowed, false otherwise.
+   */
+  static bool
+  allowDeprecatedExtensionName(absl::string_view extension_type, absl::string_view deprecated_name,
+                               absl::string_view canonical_name,
+                               Runtime::Loader* runtime = Runtime::LoaderSingleton::getExisting()) {
+    auto status = deprecatedExtensionNameStatus(runtime);
+
+    if (status == Status::Warn) {
+      ENVOY_LOG_MISC(warn, "{}", message(extension_type, deprecated_name, canonical_name));
+      return true;
+    }
```

Comment:
nit: it seems like this could be moved into `deprecatedExtensionNameStatus` since it's repeated below? nbd if not possible for some reason.

Reply:
Yeah, I can clean this up.

Reply:
So I modified Registry to use the allowDeprecatedExtensionNames function I added in this PR and was going to rearrange the utility but ended up using the non-logging status function in the LUA filter to avoid the log storm. But, that means the log message is still repeated. I could introduce another method but that's starting to feel like code golf.